### ISSUE TITLE
bug: home-manager installation deleted my `~/.config/home-manager/home.nix` file

### ISSUE THREAD

**Issue by samuela** [nix-community/home-manager#3783]
### Are you following the right branch?

- [X] My Nixpkgs and Home Manager versions are in sync

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue description

I just ran the installer off of `master` two days ago and it overwrote my perfectly customized `home.nix` config 😟 Unfortunately I did not log exactly which commit this was, but I can confirm that it did destroy my `home.nix`. In particular, I ran
```
nix-channel --add "https://github.com/nix-community/home-manager/archive/master.tar.gz" home-manager
nix-channel --update
# See https://github.com/nix-community/home-manager/issues/2564#issuecomment-994943471
export NIX_PATH="${NIX_PATH:+$NIX_PATH:}$HOME/.nix-defexpr/channels:/nix/var/nix/profiles/per-user/root/channels"
nix-shell "<home-manager>" -A install
home-manager switch
```

### System information

```markdown
- system: `"x86_64-linux"`
 - host os: `Linux 5.15.0-1030-gcp, Ubuntu, 20.04.5 LTS (Focal Fossa), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.1`
 - channels(root): `"nixpkgs"`
 - channels(sam.ainsworth): `"home-manager, nixgl"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

**Comment by ncfavier** [comment]
The default configuration path was changed in https://github.com/nix-community/home-manager/commit/cae54dc45c0d61c99c1dc8b04bc42f36c76f9771 from `~/.config/nixpkgs/home.nix` to `~/.config/home-manager/home.nix`. Maybe your old home.nix is still in the old path?

**Comment by samuela** [comment]
on further investigation i believe this was actually due to the location of `home.nix` moving from `~/.config/nixpkgs/home.nix` to `~/.config/home-manager/home.nix`. The UX here is a bit awkward. How're users supposed to know to move their configurations?

**Comment by ncfavier** [comment]
You don't have to run the installer every time you update. If you hadn't, the next `switch` would have warned you about the deprecated path.

Maybe we could add a check for `~/.config/nixpkgs/home.nix` [here](https://github.com/nix-community/home-manager/blob/3239e0b40f242f47bf6c0c37b2fd35ab3e76e370/home-manager/home-manager#L275) just in case?

**Comment by samuela** [comment]
> You don't have to run the installer every time you update.

Yeah, unfortunately I'm in a situation where I need to reinstall nix and home-manager (automatically) on every boot
